dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/AggregatedFTPolicy.java

```diff
@@ -129,14 +111,22 @@ public FallbackPolicy getFallbackPolicy() {
     }
 
     /**
+     * @param rcInstance an instance of the request context controller
      * @return
      */
     @SuppressWarnings("unchecked")
-    public Executor<Object> getExecutor() {
+    public Executor<Object> getExecutor(Instance<AsyncRequestContextController> rcInstance) {
         synchronized (this) {
             if (this.executor == null) {
                 ExecutorBuilder<?> builder = newBuilder();
 
+                // if CDI version < 2.0 then instance will be null and request context will not be active
+                if (rcInstance != null) {
+                    if (!rcInstance.isUnsatisfied() && !rcInstance.isAmbiguous()) {
+                        builder.setRequestContext(rcInstance.get());
+                    }
+                }
```

Comment:
I don't think this should ever be `null`, you'll always get an `Instance`, even if there's no bean of that type.